tests/rptest/tests/schema_registry_test.py

```diff
@@ -1251,16 +1251,14 @@ def check_subject_versions(subject: str, subject_versions: list[int]):
             res_versions = result_raw.json()
             self.logger.debug(res_versions)
             assert type(res_versions) == type([])
-            res_versions.sort()
-            subject_versions.sort()
             assert res_versions == subject_versions
 
         # Given the subject, list of schemas, list of versions, and list of ids,
         # check the schema that maps to the id
         def check_each_schema(subject: str, schemas: list[str],
                               schema_ids: list[int],
                               subject_versions: list[int]):
-            for version in subject_versions:
+            for idx, version in zip(schema_ids, subject_versions):
```

Comment:
previously if schema_ids was empty but subject_versions wasn't the test would have failed because of the assertion `assert res["id"] in schema_ids` but now we'll pass it altogether because `zip([], [1,2,3,4])` yields `[]`. it looks like the test became weaker

maybe we should do `assert len(schema_ids)==len(subject_versions)` before zipping?

Reply:
The test is already protected against an empty `schema_ids` list. The list `schema_ids` is defined in this ducktape test within the nested function `register_schemas()`. That function takes the output from `"POST /subjects/{subject}/versions`, an integer representing the schema id, and appends it into `schema_ids`.

Here is the definition of `register_schemas()`:
```py
        def register_schemas(subject: str):
            schemas = [
                json.dumps({"schema": schema1_def}),
                json.dumps({"schema": schema2_def}),
                json.dumps({"schema": schema3_def})
            ]
            schema_ids = []

            for schema in schemas:
                self.logger.debug(f"Post schema {json.loads(schema)}")
                result_raw = self._post_subjects_subject_versions(
                    subject=subject, data=schema)
                self.logger.debug(result_raw)
                assert result_raw.status_code == requests.codes.ok
                res = result_raw.json()
                self.logger.debug(res)
                assert "id" in res
                schema_ids.append(result_raw.json()["id"])

            return schemas, schema_ids
```

The way I see it, there are two pathways to get an empty `schema_ids` list:
1. If `for schema in schemas` does not execute - The test is protected against this situation since `schemas` is hardcoded in the function
2. If there is no integer returned from the call to `"POST /subjects/{subject}/versions` (`_post_subjects_subject_versions` is a wrapper around that HTTP request) - The test is protected against this situation because of `assert result_raw.status_code == requests.codes.ok` and `assert "id" in res`

Let me know if you see another pathway I may have missed.

Reply:
I've just noticed that it's a backport so it doesn't matter because we want the backport to match dev. Basically you inspected all the places and checked that the lists have the same length but it's easy to make an error and if we do it and check_each_schema won't validate its assumptions then it will take a while before we notice the issue.

I don't have a general rule up to which degree we should validate the arguments instead of trusting them but with `zip` operator it's a common mistake to pass lists of different length to it so I immediately jumped to comment. If you will touch this file again please add the assert; even if it isn't 100% necessary somebody may notice the check and decide to check too if they ever use `zip` operator.